## Manatee Mortality Stats from Florida Fish and Wildlife
### Florida Fish & Wildlife Conservation Commission Marine Mammal Pathobiology Laboratory
### 2018 Preliminary Manatee Mortality Table with 5-Year Summary From: 02/01/2018 To: 12/31/2018
### Data obtained from [Florida Fish & Wildlife Conservation Commission](https://myfwc.com/research/manatee/rescue-mortality-response/statistics/mortality/)

In [41]:
import datetime as dt
import pandas as pd

### Convert PDF to spreadsheet using [this website](https://altoconvertpdftoexcel.com/)

In [3]:
df = pd.read_excel('manatees_2018.xlsx', header=None)
df.head()

0   1   2   3   4    5   \
0  FLORIDA FISH AND WILDLIFE CONSERVATION COMMISS... NaN NaN NaN NaN  NaN   
1                                             County NaN NaN NaN NaN  NaN   
2                                             Citrus NaN NaN NaN NaN  NaN   
3                                               Levy NaN NaN NaN NaN  NaN   
4                                            Broward NaN NaN NaN NaN  NaN   

                    6    7         8    9   ...   16   17              18  \
0                  NaN  NaN       NaN  NaN  ...  NaN  NaN             NaN   
1                 Date  NaN  Field ID  NaN  ...  NaN  NaN            City   
2  2018-02-01 00:00:00  NaN  MNW18021  NaN  ...  NaN  NaN   Crystal River   
3  2018-02-01 00:00:00  NaN  MNW18026  NaN  ...  NaN  NaN       Cedar Key   
4  2018-02-01 00:00:00  NaN  MSE18024  NaN  ...  NaN  NaN  Ft. Lauderdale   

    19   20                                   21   22   23  24  25  
0  NaN  NaN                                  NaN  NaN  NaN NaN NaN  
1  NaN  NaN                       Probable Cause  NaN  NaN NaN NaN  
2  NaN  NaN                       Natural: Other  NaN  NaN NaN NaN  
3  NaN  NaN                 Natural: Cold Stress  NaN  NaN NaN NaN  
4  NaN  NaN  Human Related: Watercraft Collision  NaN  NaN NaN NaN  

[5 rows x 26 columns]

In [4]:
df.tail(20)

0   1   2   3   4   \
735                                             Walton NaN NaN NaN NaN   
736                                          St. Lucie NaN NaN NaN NaN   
737                                                Lee NaN NaN NaN NaN   
738                                             Monroe NaN NaN NaN NaN   
739                                            Collier NaN NaN NaN NaN   
740                                            Collier NaN NaN NaN NaN   
741                                                NaN NaN NaN NaN NaN   
742                                                NaN NaN NaN NaN NaN   
743                                                NaN NaN NaN NaN NaN   
744                             For The Period of Time NaN NaN NaN NaN   
745                                 Total=         824 NaN NaN NaN NaN   
746  Mortality comparisons for the time period from... NaN NaN NaN NaN   
747                                               Year NaN NaN NaN NaN   
748                                               2018 NaN NaN NaN NaN   
749                                               2017 NaN NaN NaN NaN   
750                                               2016 NaN NaN NaN NaN   
751                                               2015 NaN NaN NaN NaN   
752                                               2014 NaN NaN NaN NaN   
753                                               2013 NaN NaN NaN NaN   
754                                     5-year average NaN NaN NaN NaN   

             5                    6                7         8   \
735         NaN  2018-12-30 00:00:00              NaN  MNW18166   
736         NaN  2018-12-30 00:00:00              NaN  MSE18098   
737         NaN  2018-12-30 00:00:00              NaN  MSW18340   
738         NaN  2018-12-30 00:00:00              NaN  MSW18341   
739         NaN  2018-12-30 00:00:00              NaN  MSW18343   
740         NaN  2018-12-31 00:00:00              NaN  MSW18342   
741         NaN                  NaN              NaN       NaN   
742         NaN                  NaN              NaN       NaN   
743         NaN                  NaN              NaN       NaN   
744         NaN                  NaN              NaN       NaN   
745         NaN                  NaN              NaN       NaN   
746         NaN                  NaN              NaN       NaN   
747  Watercraft                  NaN  Flood Gate/Lock       NaN   
748   123 (15%)                  NaN                6       NaN   
749   111 (21%)                  NaN                3       NaN   
750   106 (20%)                  NaN                2       NaN   
751    86 (21%)                  NaN                5       NaN   
752    69 (19%)                  NaN                3       NaN   
753     73 (9%)                  NaN                5       NaN   
754    89 (18%)                  NaN                3       NaN   

                           9   ...           16       17          18  \
735                       NaN  ...          NaN      NaN   Niceville   
736                       NaN  ...          NaN      NaN  Ft. Pierce   
737                       NaN  ...          NaN      NaN  Cape Coral   
738                       NaN  ...          NaN      NaN    Flamingo   
739                       NaN  ...          NaN      NaN     Ochopee   
740                       NaN  ...          NaN      NaN      Naples   
741                       NaN  ...          NaN      NaN         NaN   
742                       NaN  ...          NaN      NaN         NaN   
743                       NaN  ...          NaN      NaN         NaN   
744  01/01/2018 -\n12/31/2018  ...          NaN      NaN         NaN   
745                       NaN  ...          NaN      NaN         NaN   
746                       NaN  ...          NaN      NaN         NaN   
747                       NaN  ...  Cold Stress  Natural         NaN   
748                       NaN  ...           79      228         NaN   
749      

In [5]:
# test slice

df.iloc[749:753, 0:]

0   1   2   3   4          5    6  7    8    9   ...  16  17   18  19  \
749  2017 NaN NaN NaN NaN  111 (21%)  NaN  3  NaN  NaN  ...  30  93  NaN  26   
750  2016 NaN NaN NaN NaN  106 (20%)  NaN  2  NaN  NaN  ...  24  86  NaN  36   
751  2015 NaN NaN NaN NaN   86 (21%)  NaN  5  NaN  NaN  ...  18  44  NaN  31   
752  2014 NaN NaN NaN NaN   69 (19%)  NaN  3  NaN  NaN  ...  26  29  NaN  17   

      20   21  22   23  24  25  
749  114  NaN  29  538 NaN NaN  
750  113  NaN  34  520 NaN NaN  
751   95  NaN  24  405 NaN NaN  
752   80  NaN  38  371 NaN NaN  

[4 rows x 26 columns]

In [6]:
clean_df = df.iloc[:741, 0:]
clean_df.tail(10)

0   1   2   3   4    5                    6    7         8   \
731  Hillsborough NaN NaN NaN NaN  NaN  2018-12-27 00:00:00  NaN  MNW18164   
732  Indian River NaN NaN NaN NaN  NaN  2018-12-28 00:00:00  NaN  MEC18150   
733  Hillsborough NaN NaN NaN NaN  NaN  2018-12-29 00:00:00  NaN  MNW18165   
734       Broward NaN NaN NaN NaN  NaN  2018-12-29 00:00:00  NaN  MSE18097   
735        Walton NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MNW18166   
736     St. Lucie NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MSE18098   
737           Lee NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MSW18340   
738        Monroe NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MSW18341   
739       Collier NaN NaN NaN NaN  NaN  2018-12-30 00:00:00  NaN  MSW18343   
740       Collier NaN NaN NaN NaN  NaN  2018-12-31 00:00:00  NaN  MSW18342   

      9   ...   16   17            18   19   20  \
731  NaN  ...  NaN  NaN  Apollo Beach  NaN  NaN   
732  NaN  ...  NaN  NaN     Sebastian  NaN  NaN   
733  NaN  ...  NaN  NaN         Tampa  NaN  NaN   
734  NaN  ...  NaN  NaN     Hollywood  NaN  NaN   
735  NaN  ...  NaN  NaN     Niceville  NaN  NaN   
736  NaN  ...  NaN  NaN    Ft. Pierce  NaN  NaN   
737  NaN  ...  NaN  NaN    Cape Coral  NaN  NaN   
738  NaN  ...  NaN  NaN      Flamingo  NaN  NaN   
739  NaN  ...  NaN  NaN       Ochopee  NaN  NaN   
740  NaN  ...  NaN  NaN        Naples  NaN  NaN   

                                      21   22   23  24  25  
731                 Human Related: Other  NaN  NaN NaN NaN  
732         Undetermined: Too Decomposed  NaN  NaN NaN NaN  
733                       Natural: Other  NaN  NaN NaN NaN  
734         Undetermined: Too Decomposed  NaN  NaN NaN NaN  
735                 Natural: Cold Stress  NaN  NaN NaN NaN  
736         Undetermined: Too Decomposed  NaN  NaN NaN NaN  
737                Perinatal (<= 150 cm)  NaN  NaN NaN NaN  
738  Human Related: Watercraft Collision  NaN  NaN NaN NaN  
739                 Natural: Cold Stress  NaN  NaN NaN NaN  
740         Undetermined: Too Decomposed  NaN  NaN NaN NaN  

[10 rows x 26 columns]

#### Drop NA columns

In [7]:
clean_df.isnull().sum()

0       0
1     741
2     741
3     741
4     741
5     741
6       1
7     741
8       1
9     741
10    741
11    741
12      1
13      3
14    741
15      1
16    741
17    741
18      1
19    741
20    741
21      1
22    741
23    741
24    741
25    741
dtype: int64

In [8]:
# drop columns with a threshold of more than 3 NA values because I'm not sure what the column 13 is about

clean_df.dropna(axis=1, thresh=3, inplace=True)
clean_df.head()

0                    6   \
0  FLORIDA FISH AND WILDLIFE CONSERVATION COMMISS...                  NaN   
1                                             County                 Date   
2                                             Citrus  2018-02-01 00:00:00   
3                                               Levy  2018-02-01 00:00:00   
4                                            Broward  2018-02-01 00:00:00   

         8    12         13                 15              18  \
0       NaN  NaN        NaN                NaN             NaN   
1  Field ID  Sex  Size (cm)           Waterway            City   
2  MNW18021    M        269          Kings Bay   Crystal River   
3  MNW18026    M        202  Black Point Swamp       Cedar Key   
4  MSE18024    U        275    Port Everglades  Ft. Lauderdale   

                                    21  
0                                  NaN  
1                       Probable Cause  
2                       Natural: Other  
3                 Natural: Cold Stress  
4  Human Related: Watercraft Collision

In [9]:
# drop first row and convert second row to headers

clean_df.drop(df.index[0], inplace=True)
clean_df.head()

0                    6         8    12         13                 15  \
1   County                 Date  Field ID  Sex  Size (cm)           Waterway   
2   Citrus  2018-02-01 00:00:00  MNW18021    M        269          Kings Bay   
3     Levy  2018-02-01 00:00:00  MNW18026    M        202  Black Point Swamp   
4  Broward  2018-02-01 00:00:00  MSE18024    U        275    Port Everglades   
5  Brevard  2018-02-02 00:00:00  MEC18013    F        280       Indian River   

               18                                   21  
1            City                       Probable Cause  
2   Crystal River                       Natural: Other  
3       Cedar Key                 Natural: Cold Stress  
4  Ft. Lauderdale  Human Related: Watercraft Collision  
5   Port St. John                 Human Related: Other

In [10]:
# convert headers

headers = clean_df.iloc[0]
new_df = pd.DataFrame(data=clean_df.values[1:], columns=headers)
new_df.head()

1   County                 Date  Field ID Sex Size (cm)           Waterway  \
0   Citrus  2018-02-01 00:00:00  MNW18021   M       269          Kings Bay   
1     Levy  2018-02-01 00:00:00  MNW18026   M       202  Black Point Swamp   
2  Broward  2018-02-01 00:00:00  MSE18024   U       275    Port Everglades   
3  Brevard  2018-02-02 00:00:00  MEC18013   F       280       Indian River   
4  Brevard  2018-02-02 00:00:00  MEC18014   M       209   Eau Gallie River   

1            City                       Probable Cause  
0   Crystal River                       Natural: Other  
1       Cedar Key                 Natural: Cold Stress  
2  Ft. Lauderdale  Human Related: Watercraft Collision  
3   Port St. John                 Human Related: Other  
4       Melbourne                 Natural: Cold Stress

In [28]:
# I'm going to remove that unnecessary timestamp eventually

In [42]:
manatee_df = pd.read_csv('manatee_mortality_2018.csv')
manatee_df.head()

County                 Date  Field ID Sex Size (cm)           Waterway  \
0   Citrus  2018-02-01 00:00:00  MNW18021   M       269          Kings Bay   
1     Levy  2018-02-01 00:00:00  MNW18026   M       202  Black Point Swamp   
2  Broward  2018-02-01 00:00:00  MSE18024   U       275    Port Everglades   
3  Brevard  2018-02-02 00:00:00  MEC18013   F       280       Indian River   
4  Brevard  2018-02-02 00:00:00  MEC18014   M       209   Eau Gallie River   

             City                       Probable Cause  
0   Crystal River                       Natural: Other  
1       Cedar Key                 Natural: Cold Stress  
2  Ft. Lauderdale  Human Related: Watercraft Collision  
3   Port St. John                 Human Related: Other  
4       Melbourne                 Natural: Cold Stress

In [43]:
manatee_df.tail()

County                 Date  Field ID Sex Size (cm)       Waterway  \
734  St. Lucie  2018-12-30 00:00:00  MSE18098   M       201   Taylor Creek   
735        Lee  2018-12-30 00:00:00  MSW18340   F       140  Matlacha Pass   
736     Monroe  2018-12-30 00:00:00  MSW18341   M       265    Florida Bay   
737    Collier  2018-12-30 00:00:00  MSW18343   M       208  Tamiami Canal   
738    Collier  2018-12-31 00:00:00  MSW18342   F       154    Roberts Bay   

           City                       Probable Cause  
734  Ft. Pierce         Undetermined: Too Decomposed  
735  Cape Coral                Perinatal (<= 150 cm)  
736    Flamingo  Human Related: Watercraft Collision  
737     Ochopee                 Natural: Cold Stress  
738      Naples         Undetermined: Too Decomposed

## Probable Causes of Death
### Official Categories & Descriptions
* __Watercraft__ - Watercraft-related mortality. Death may result from propeller wounds, impact, crushing, or any combination of the three.
* __Crush / Drowned in Flood Gate or Canal Lock__ - Manatees killed by crushing or asphyxiation in flood gates and canal locks.
* __Other Human Related__ - Manatee deaths caused by vandalism, poaching, entrapment in pipes and culverts, complications due to entanglement in ropes, lines, and nets, or ingestion of fishing gear or debris.
* __Perinatal__ - Manatees less than or equal to 150 cm (5 feet) in total length which were not determined to have died due to human-related causes.
* __Cold Stress__ - Manatees which die as a result of exposure to acute or prolonged cold weather. Animals are usually emaciated and in a general state of malnutrition. (Combined with "Other Natural" in some printouts.)
* __Other Natural__ - Manatee deaths resulting from infectious and non-infectious diseases, birth complications, natural accidents, and natural catastrophes (such as red tide blooms).
* __Undetermined: Too Decomposed__ - Manatee deaths in which the cause of death could not be determined due to an advanced stage of decomposition.
* __Undetermined: Other__ - Manatee deaths in which the carcass was in good condition (fresh or moderately decomposed), but necropsy findings were inconclusive. (Combined with "Undetermined" in some printouts.)
* __Verified / Not Recovered__ - Manatee deaths that were reported and verified, but a complete necropsy was not performed. (Combined with "Undetermined" in some printouts.)

In [44]:
manatee_df['Probable Cause'].value_counts()

Natural: Other                          213
Human Related: Watercraft Collision     110
Undetermined: Too Decomposed            108
Perinatal (<= 150 cm)                   107
Verified: Not Recovered                  97
Natural: Cold Stress                     37
Undetermined: Other                      27
Probable Cause                           19
Human Related: Other                     16
Human Related: Flood Gate/Canal Lock      5
Name: Probable Cause, dtype: int64

#### Can drop rows 744 & 745 because they're just a summary of the deaths for all of 2018:
* 824 total deaths
* 123 Watercraft Deaths (Florida only)

### Mortality comparisons for the time period from 01/01 to the ending date (final + preliminary data)


In [31]:
comparison_df = df.iloc[747:, 0:]
comparison_df

0   1   2   3   4           5    6                7    8   \
747            Year NaN NaN NaN NaN  Watercraft  NaN  Flood Gate/Lock  NaN   
748            2018 NaN NaN NaN NaN   123 (15%)  NaN                6  NaN   
749            2017 NaN NaN NaN NaN   111 (21%)  NaN                3  NaN   
750            2016 NaN NaN NaN NaN   106 (20%)  NaN                2  NaN   
751            2015 NaN NaN NaN NaN    86 (21%)  NaN                5  NaN   
752            2014 NaN NaN NaN NaN    69 (19%)  NaN                3  NaN   
753            2013 NaN NaN NaN NaN     73 (9%)  NaN                5  NaN   
754  5-year average NaN NaN NaN NaN    89 (18%)  NaN                3  NaN   

      9   ...           16       17   18           19            20   21  \
747  NaN  ...  Cold Stress  Natural  NaN  Unrecovered  Undetermined  NaN   
748  NaN  ...           79      228  NaN          105           120  NaN   
749  NaN  ...           30       93  NaN           26           114  NaN   
750  NaN  ...           24       86  NaN           36           113  NaN   
751  NaN  ...           18       44  NaN           31            95  NaN   
752  NaN  ...           26       29  NaN           17            80  NaN   
753  NaN  ...           40      198  NaN          100           125  NaN   
754  NaN  ...           27       90  NaN           42           105  NaN   

                     22     23  24  25  
747  Undetermined Other  Total NaN NaN  
748                  29    824 NaN NaN  
749                  29    538 NaN NaN  
750                  34    520 NaN NaN  
751                  24    405 NaN NaN  
752                  38    371 NaN NaN  
753                 148    830 NaN NaN  
754                  54    532 NaN NaN  

[8 rows x 26 columns]

In [32]:
comparison_df.isnull().sum()

0     0
1     8
2     8
3     8
4     8
5     0
6     8
7     0
8     8
9     8
10    8
11    0
12    8
13    8
14    0
15    8
16    0
17    0
18    8
19    0
20    0
21    8
22    0
23    0
24    8
25    8
dtype: int64

In [34]:
df.iloc[747:, 8:]

8    9    10           11   12   13         14   15           16  \
747  NaN  NaN  NaN  Other Human  NaN  NaN  Perinatal  NaN  Cold Stress   
748  NaN  NaN  NaN           21  NaN  NaN        113  NaN           79   
749  NaN  NaN  NaN           15  NaN  NaN        117  NaN           30   
750  NaN  NaN  NaN            6  NaN  NaN        113  NaN           24   
751  NaN  NaN  NaN           11  NaN  NaN         91  NaN           18   
752  NaN  NaN  NaN            9  NaN  NaN        100  NaN           26   
753  NaN  NaN  NaN           12  NaN  NaN        129  NaN           40   
754  NaN  NaN  NaN           10  NaN  NaN        110  NaN           27   

          17   18           19            20   21                  22     23  \
747  Natural  NaN  Unrecovered  Undetermined  NaN  Undetermined Other  Total   
748      228  NaN          105           120  NaN                  29    824   
749       93  NaN           26           114  NaN                  29    538   
750       86  NaN           36           113  NaN                  34    520   
751       44  NaN           31            95  NaN                  24    405   
752       29  NaN           17            80  NaN                  38    371   
753      198  NaN          100           125  NaN                 148    830   
754       90  NaN           42           105  NaN                  54    532   

     24  25  
747 NaN NaN  
748 NaN NaN  
749 NaN NaN  
750 NaN NaN  
751 NaN NaN  
752 NaN NaN  
753 NaN NaN  
754 NaN NaN

In [35]:
# drop null columns
comparison_df.dropna(axis=1, inplace=True)
comparison_df

0           5                7            11         14  \
747            Year  Watercraft  Flood Gate/Lock  Other Human  Perinatal   
748            2018   123 (15%)                6           21        113   
749            2017   111 (21%)                3           15        117   
750            2016   106 (20%)                2            6        113   
751            2015    86 (21%)                5           11         91   
752            2014    69 (19%)                3            9        100   
753            2013     73 (9%)                5           12        129   
754  5-year average    89 (18%)                3           10        110   

              16       17           19            20                  22  \
747  Cold Stress  Natural  Unrecovered  Undetermined  Undetermined Other   
748           79      228          105           120                  29   
749           30       93           26           114                  29   
750           24       86           36           113                  34   
751           18       44           31            95                  24   
752           26       29           17            80                  38   
753           40      198          100           125                 148   
754           27       90           42           105                  54   

        23  
747  Total  
748    824  
749    538  
750    520  
751    405  
752    371  
753    830  
754    532

In [36]:
comparison_df.to_csv('five_year_mortality_comparison.csv', index=False)

## Load & Inspect Other Years
* In 2017, the West Indian Manatee Conservation Status was downgraded from __endangered__ to __threatened__.

### Inspect xlsx files to write custom fuction to clean data

In [50]:
mortality_2017 = pd.read_excel('manatees_2017.xlsx', header=None)
mortality_2017.head()

0           1   \
0  FLORIDA FISH AND WILDLIFE CONSERVATION COMMISS...         NaN   
1                                             County  Watercraft   
2                                                Bay           0   
3                                            Brevard          13   
4                                            Broward           7   

                       2            3          4            5        6   \
0                     NaN          NaN        NaN          NaN      NaN   
1  Flood Gate/ Canal Lock  Other Human  Perinatal  Cold Stress  Natural   
2                       0            0          0            0        0   
3                       0            4         21           13        5   
4                       0            0          0            0        2   

             7            8      9   10  
0           NaN          NaN    NaN NaN  
1  Undetermined  Unrecovered  Total NaN  
2             1            2      3 NaN  
3            54            1    111 NaN  
4             4            0     13 NaN

In [51]:
mortality_2016 = pd.read_excel('manatees_2016.xlsx', header=None)
mortality_2016.head()

0           1   \
0  FLORIDA FISH AND WILDLIFE CONSERVATION COMMISS...         NaN   
1                                             County  Watercraft   
2                                                Bay           0   
3                                            Brevard          10   
4                                            Broward           5   

                       2            3          4            5        6   \
0                     NaN          NaN        NaN          NaN      NaN   
1  Flood Gate/ Canal Lock  Other Human  Perinatal  Cold Stress  Natural   
2                       0            0          0            0        0   
3                       0            1         45            4        3   
4                       0            0          1            0        3   

             7            8      9   10  
0           NaN          NaN    NaN NaN  
1  Undetermined  Unrecovered  Total NaN  
2             0            1      1 NaN  
3            41            6    110 NaN  
4             9            0     18 NaN

In [52]:
mortality_2019 = pd.read_excel('manatees_2019.xlsx', header=None)
mortality_2019.head()

0   1   2   3   4    5   \
0  FLORIDA FISH AND WILDLIFE CONSERVATION COMMISS... NaN NaN NaN NaN  NaN   
1                                             County NaN NaN NaN NaN  NaN   
2                                             Nassau NaN NaN NaN NaN  NaN   
3                                       Hillsborough NaN NaN NaN NaN  NaN   
4                                             Monroe NaN NaN NaN NaN  NaN   

                    6    7         8    9   ...                15   16   17  \
0                  NaN  NaN       NaN  NaN  ...               NaN  NaN  NaN   
1                 Date  NaN  Field ID  NaN  ...          Waterway  NaN  NaN   
2  2019-01-01 00:00:00  NaN  MNE19001  NaN  ...      Nassau River  NaN  NaN   
3  2019-01-01 00:00:00  NaN  MNW19001  NaN  ...  Hillsborough Bay  NaN  NaN   
4  2019-01-01 00:00:00  NaN  MSW19001  NaN  ...       Florida Bay  NaN  NaN   

             18   19   20                    21   22   23  24  
0           NaN  NaN  NaN                   NaN  NaN  NaN NaN  
1          City  NaN  NaN        Probable Cause  NaN  NaN NaN  
2         Yulee  NaN  NaN  Natural: Cold Stress  NaN  NaN NaN  
3  Apollo Beach  NaN  NaN  Natural: Cold Stress  NaN  NaN NaN  
4      Flamingo  NaN  NaN   Undetermined: Other  NaN  NaN NaN  

[5 rows x 25 columns]

#### If spreadsheet files are examined and have row 0 for title and row 1 for headers, could write custom fuction to take in particular row numbers, depending on the file.

In [53]:
mortality_2019.iloc[0,0]

'FLORIDA FISH AND WILDLIFE CONSERVATION COMMISSION\nMARINE MAMMAL PATHOBIOLOGY LABORATORY\n2019 Preliminary Manatee Mortality Table with 5-Year Summary From:  01/01/2019    To:  07/19/2019'

In [56]:
mortality_2019.tail(15)

0   1   2   3   4   \
350                                                Lee NaN NaN NaN NaN   
351                                                NaN NaN NaN NaN NaN   
352                                                NaN NaN NaN NaN NaN   
353                                                NaN NaN NaN NaN NaN   
354                             For The Period of Time NaN NaN NaN NaN   
355                                 Total=         340 NaN NaN NaN NaN   
356  Mortality comparisons for the time period from... NaN NaN NaN NaN   
357                                               Year NaN NaN NaN NaN   
358                                               2019 NaN NaN NaN NaN   
359                                               2018 NaN NaN NaN NaN   
360                                               2017 NaN NaN NaN NaN   
361                                               2016 NaN NaN NaN NaN   
362                                               2015 NaN NaN NaN NaN   
363                                               2014 NaN NaN NaN NaN   
364                                     5-year average NaN NaN NaN NaN   

             5                    6                7         8   \
350         NaN  2019-07-19 00:00:00              NaN  MSW19117   
351         NaN                  NaN              NaN       NaN   
352         NaN                  NaN              NaN       NaN   
353         NaN                  NaN              NaN       NaN   
354         NaN                  NaN              NaN       NaN   
355         NaN                  NaN              NaN       NaN   
356         NaN                  NaN              NaN       NaN   
357  Watercraft                  NaN  Flood Gate/Lock       NaN   
358    93 (27%)                  NaN                2       NaN   
359    73 (15%)                  NaN                2       NaN   
360    59 (16%)                  NaN                0       NaN   
361    67 (20%)                  NaN                1       NaN   
362    57 (20%)                  NaN                4       NaN   
363    45 (19%)                  NaN                1       NaN   
364    60 (18%)                  NaN                1       NaN   

                           9   ...          15           16       17  \
350                       NaN  ...  Estero Bay          NaN      NaN   
351                       NaN  ...         NaN          NaN      NaN   
352                       NaN  ...         NaN          NaN      NaN   
353                       NaN  ...         NaN          NaN      NaN   
354  01/01/2019 -\n07/19/2019  ...         NaN          NaN      NaN   
355                       NaN  ...         NaN          NaN      NaN   
356                       NaN  ...         NaN          NaN      NaN   
357                       NaN  ...         NaN  Cold Stress  Natural   
358                       NaN  ...         NaN           39       38   
359                       NaN  ...         NaN           72      112   
360                       NaN  ...         NaN           25       79   
361                       NaN  ...         NaN           23       50   
362                       NaN  ...         NaN           18       25   
363                       NaN  ...         NaN           19       14   
364                       NaN  ...         NaN           31       56   

                  18           19            20                            21  \
350  Ft. Myers Beach          NaN           NaN  Undetermined: Too Decomposed   
351              NaN          NaN           NaN                           NaN   
352              NaN          NaN           NaN                           NaN   
353              NaN          NaN           NaN                           NaN   
354              NaN          NaN           NaN                           NaN   
355              NaN          NaN           NaN                           NaN   
356              NaN          NaN           NaN                           

In [58]:
mortality_2019.iloc[354:]

0   1   2   3   4   \
354                             For The Period of Time NaN NaN NaN NaN   
355                                 Total=         340 NaN NaN NaN NaN   
356  Mortality comparisons for the time period from... NaN NaN NaN NaN   
357                                               Year NaN NaN NaN NaN   
358                                               2019 NaN NaN NaN NaN   
359                                               2018 NaN NaN NaN NaN   
360                                               2017 NaN NaN NaN NaN   
361                                               2016 NaN NaN NaN NaN   
362                                               2015 NaN NaN NaN NaN   
363                                               2014 NaN NaN NaN NaN   
364                                     5-year average NaN NaN NaN NaN   

             5    6                7    8                         9   ...  \
354         NaN  NaN              NaN  NaN  01/01/2019 -\n07/19/2019  ...   
355         NaN  NaN              NaN  NaN                       NaN  ...   
356         NaN  NaN              NaN  NaN                       NaN  ...   
357  Watercraft  NaN  Flood Gate/Lock  NaN                       NaN  ...   
358    93 (27%)  NaN                2  NaN                       NaN  ...   
359    73 (15%)  NaN                2  NaN                       NaN  ...   
360    59 (16%)  NaN                0  NaN                       NaN  ...   
361    67 (20%)  NaN                1  NaN                       NaN  ...   
362    57 (20%)  NaN                4  NaN                       NaN  ...   
363    45 (19%)  NaN                1  NaN                       NaN  ...   
364    60 (18%)  NaN                1  NaN                       NaN  ...   

      15           16       17   18           19            20   21  \
354  NaN          NaN      NaN  NaN          NaN           NaN  NaN   
355  NaN          NaN      NaN  NaN          NaN           NaN  NaN   
356  NaN          NaN      NaN  NaN          NaN           NaN  NaN   
357  NaN  Cold Stress  Natural  NaN  Unrecovered  Undetermined  NaN   
358  NaN           39       38  NaN           41            62  NaN   
359  NaN           72      112  NaN           48            71  NaN   
360  NaN           25       79  NaN           17            75  NaN   
361  NaN           23       50  NaN           21            65  NaN   
362  NaN           18       25  NaN           20            63  NaN   
363  NaN           19       14  NaN           11            54  NaN   
364  NaN           31       56  NaN           23            65  NaN   

                     22     23  24  
354                 NaN    NaN NaN  
355                 NaN    NaN NaN  
356                 NaN    NaN NaN  
357  Undetermined Other  Total NaN  
358                  15    340 NaN  
359                  24    483 NaN  
360                  16    364 NaN  
361                  26    332 NaN  
362                  20    279 NaN  
363                  29    242 NaN  
364                  23    340 NaN  

[11 rows x 25 columns]

In [68]:
# Function that takes in title (year) of data, new title, rows for summary data, new title for summary data, headers, and 
# drops na columns with a header of na?
# Summary df format may have to be dealt with on an individual basis

def clean_spreadsheeet_manatee_file(spreadhseet_df, new_title_main_df, beginning_row_for_summary, last_row_main_df, \
                                    summary_df_name):
    
    summary_df_name = spreadsheet_df.iloc[:beginning_row_for_summary]
    temp_df = spreadsheet_df.iloc[1:, :last_row_main_df]
    temp_df.drop('NaN', 1, inplace=True)
    new_title_main_df = pd.DataFrame(data=temp_df, columns=headers)                              

In [62]:
# test this crazy function

In [64]:
mortality_2016.head()

0           1   \
0  FLORIDA FISH AND WILDLIFE CONSERVATION COMMISS...         NaN   
1                                             County  Watercraft   
2                                                Bay           0   
3                                            Brevard          10   
4                                            Broward           5   

                       2            3          4            5        6   \
0                     NaN          NaN        NaN          NaN      NaN   
1  Flood Gate/ Canal Lock  Other Human  Perinatal  Cold Stress  Natural   
2                       0            0          0            0        0   
3                       0            1         45            4        3   
4                       0            0          1            0        3   

             7            8      9   10  
0           NaN          NaN    NaN NaN  
1  Undetermined  Unrecovered  Total NaN  
2             0            1      1 NaN  
3            41            6    110 NaN  
4             9            0     18 NaN

In [66]:
mortality_2016.tail(20)

0    1  2  3    4   5   6    7   8    9   10
16        Lake    0  0  0    1   0   0    0   0    1 NaN
17         Lee   19  0  1   11   3  17   25   8   84 NaN
18        Levy    0  0  0    1   1   0    1   1    4 NaN
19     Manatee    4  0  0    2   1   6    2   0   15 NaN
20      Marion    1  0  0    0   0   0    0   0    1 NaN
21      Martin    3  0  0    4   1   2    1   0   11 NaN
22  Miami-Dade    4  1  1    1   0   0    5   2   14 NaN
23      Monroe   10  0  0    0   0   3    6   5   24 NaN
24  Okeechobee    0  0  0    1   1   0    1   0    3 NaN
25      Orange    0  0  0    0   0   0    1   0    1 NaN
26  Palm Beach    4  0  0    1   1   0    2   0    8 NaN
27       Pasco    1  0  1    1   0   2    0   0    5 NaN
28    Pinellas    4  0  0    7   1   5    6   0   23 NaN
29      Putnam    3  0  0    1   0   0    4   0    8 NaN
30    Sarasota    2  0  0    6   0   8    3   1   20 NaN
31    Seminole    0  0  0    0   0   0    1   1    2 NaN
32   St. Johns    3  0  0    2   1   0    1   0    7 NaN
33   St. Lucie    1  0  1    1   0   0    1   0    4 NaN
34     Volusia   12  0  0   11   1   3    5   1   33 NaN
35      TOTAL:  106  2  6  113  24  86  147  36  520 NaN

In [67]:
# okay. . . well 2016 does not appear to have a summary table at the end, but we'll try it anyway

In [69]:
clean_spreadsheeet_manatee_file(mortality_2016, mortality_2016_df, 35, 34, summary_2016)

NameError: name 'mortality_2016_df' is not defined